findspam.py

```diff
@@ -794,6 +794,34 @@ def misleading_link(s, site):
         return False, ''
 
 
+# noinspection PyUnusedLocal,PyMissingTypeHints,PyTypeChecker
+@create_rule("text repeated in {}", title=False, body_summary=True, max_rep=10000, max_score=10000)
+def body_text_repeated(s, site):
+    """
+    Do some hacks to reduce the need for regex backtracking for this rule
+    """
+    s = s.rstrip("\n")
+    if s.startswith("<p>") and s.endswith("</p>"):
+        s = s[3:-4]
+    initial_words = regex.match(r"\A([^\W_]+)[\W_]+([^\W_]+)[\W_]+([^\W_]+)", s)
+    if not initial_words:
+        return False, ""
+    escaped_initial_words = [regex.escape(x) for x in initial_words.groups()]
+    period = regex.match(
```

Comment:
Please add `regex.cache_all(False)` prior to this `regex.match()` and `regex.cache_all(True)` after it to prevent the single-use regexes compiled here from filling the `regex` package's internal cache of compiled regular expressions.

Note: this is probably something which we should be doing in a substantial number of places in the code, but I was reminded when creating PR #7012 that it should be used here. Basically, we should be doing that wherever we use a regex which is unique per-post. We should also change things in individual functions/modules to separately store compiled regexes which are used repeatedly in order to reduce the  number which we rely on the `regex` package to cache.

Reply:
I've struck-out my earlier comments here and closed an open PR I had, as `regex.cache_all()` is the wrong way to handle this here, or anywhere in SD, due to threading. I'll be writing a helper function which can be used for explicit regex compiles to force the implicit `regex` cache to not be used. I'll update this again once I've created that (soon). I'll also create an issue in the `regex` package asking to expose an option to be able to not use the implicit cache on a per-compile basis.

Reply:
I've opened a PR, #7025, which adds `regex_compile_no_cache()` to helpers.py and imports it to findspam.py. That function can be used to compile a regular expression and not have it placed in the `regex` package's implicit cache. Using that function, the code here could be something like:
```
    period_regex = regex_compile_no_cache(
        r"\A%s[\W_]+%s[\W_]+%s[\W_]+(.{1,40}?)%s[\W_]+%s[\W_]+%s(?=$|[\W_])" % (
            tuple(escaped_initial_words * 2))
    period = period_regex.match(s)
    if not period:
        return False, ""
    period_words = regex.split(r"[\W_]+", period.groups(0)[0])
    escaped_words = escaped_initial_words + [
        regex.escape(x) for x in period_words]
    repeats_regex = regex_compile_no_cache(r"\A(" + r"[\W_]+".join(escaped_words) + r"[\W_]*){10,}")
    repeats = repeats_regex.match(s)
```